### PR TITLE
doc: add usage and configuration docs

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

File: doc/ignition.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## Configuration ##
+
+The Ignition configuration is provided in a JSON document via one of the
+aforementioned config providers. The format of the document is detailed in the
+following section.
+
+### Specification ###
+
+- **version** (integer): the version number of the spec. Must be `1`.
```
Comment (vcaputo) [inline doc/ignition.md]: These should all be capitalized after the colon.

File: doc/ignition.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+- **storage** (object): describes the desired state of the systemd's storage
```
Comment (vcaputo) [inline doc/ignition.md]: s/systemd/system/

File: doc/ignition.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                        devices.
+  - **disks** (list of objects): the list of disks to be configured and their
+                                 options.
+    - **device** (string): the absolute path to the device. Devices are
+                           typically referenced by the /dev/disk/by-* symlinks.
+    - **wipe-table** (boolean): when true, the partition tables are erased
+                                before any further manipulation. Otherwise, the
+                                existing entries are left in-tact.
```
Comment (vcaputo) [inline doc/ignition.md]: intact is a word.

File: doc/ignition.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    - **partitions** (list of objects): the list of partitions and their
+                                        configuration for this particular disk.
+      - **label** (string): the PARTLABEL for the partition.
+      - **number** (integer): the partition number, which dictates it's
+                              position in the partition table.
+      - **size** (integer): the size of the partition (in sectors).
+      - **start** (integer): the start of the partition (in sectors).
+      - **type-guid** (string): the GPT [partition type GUID][part-types].
+  - **raid** (list of objects): the list of RAID arrays to be configured.
+    - **name** (string): ?
```
Comment (vcaputo) [inline doc/ignition.md]: "The name to use for the resulting md device." ?

File: doc/ignition.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                                     absolute path) in the array.
+    - **spares** (integer): the number of spares (if applicable) in the array.
+  - **filesystems** (list of objects): the list of filesystems to be
+                                       configured. Typically, one filesystem
+                                       is configured per partition.
+    - **device** (string): the absolute path to the device. Devices are
+                           typically referenced by the /dev/disk/by-* symlinks.
+    - **initialize** (boolean): when true, any existing existing data on the
+                                partition is destroyed during the formatting.
+                                Otherwise, if an existing filesystem is
```
Comment (vcaputo) [inline doc/ignition.md]: this isn't how it's implemented today.

If true, we initialize the filesystem.  If false, we don't do any initialization here but will mount an existing filesystem if you've included files destined for this filesystem.

File: doc/ignition.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                                detected, the format operation is aborted and
+                                the setup fails leaving the data in-tact.
```
Comment (vcaputo) [inline doc/ignition.md]: intact

File: doc/ignition.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    - **format** (string): the filesystem format (e.g. ext4, btrfs, etc.).
+    - **options** (list of strings): any additional options to be passed to
+                                     the sgdisk utility.
```
Comment (vcaputo) [inline doc/ignition.md]: format-specific mkfs utility.

File: doc/ignition.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    - **files** (list of objects): the list of files, rooted in this particular
+                                   filesystem, to be written.
+      - **path** (string): the absolute path to the file.
+      - **contents** (string): the contents of the file.
+      - **mode** (integer): the file's permission mode. Note that the mode must
+                            be properly specified as a **decimal** value
+                            (i.e. 0644 -> 420).
```
Comment (vcaputo) [inline doc/ignition.md]: this... argh.

**Comment by vcaputo** [comment]
some stuff, otherwise lgtm, good start.

File: doc/ignition.md (hunk 1 above)
Comment (crawford) [inline doc/ignition.md]: Hmmmmmm, my english isn't great, but I don't think those should be capitalized.

Comment (vcaputo) [inline doc/ignition.md]: fine

**Comment by vcaputo** [comment]
lgtm

**Pull request merged**